connections/README.md

```diff
@@ -263,6 +267,49 @@ indicating whether the handler supports the protocol. This allows more flexible
 behavior than exact literal matching, which is the default behavior if no match
 function is provided.
 
+## Stream Behaviour
+
+This section specifies the expected behaviour of a libp2p stream.
+
+### States
+
+A libp2p stream can be in one of four states:
+
+- Open for reading and writing
+- Open for reading only
+- Open for writing only
+- Closed
+
+The default state after opening a new stream is "Open for reading and writing".
+Either party can at any point close the reading or writing side.
+
+The "Closed" state is the product of both halfs (read and write) of the stream being closed.
+This can happen because one party closes both halfs but also as a combination
+of both parties closing their write-half (which automatically closes other's read-half).
+
+Writing to a write-closed stream SHOULD fail with an error clearly signalling the closed state.
+
+Reading from a read-closed stream SHOULD succeed by reading 0 bytes.
+This allows for a robust "read to end"-behaviour.
+Layers on top of the stream MAY fail with an unexpected EOF error in case more data was expected.
```

Comment:
I don't think this needs to be in the specification.

Reply:
Would you just omit it? I think I'd find it helpful because it explains, _why_ we want read operations on closed streams to be successful. Without this, people may rightfully ask, why reading from a read-closed stream shouldn't fail?

I am happy to reword it, but I think it is useful to have in the spec in one form or another.